docs/source/howto/forced-login.md

```diff
@@ -0,0 +1,129 @@
+Step 2 and 3 use the JupyterHub [REST API](#jupyterhub-rest-API).
+The service would need the scopes:
+
+```
+admin:users # creating users
+servers # start/stop servers
+```
+
+That leaves the last step: sending users to their running server with credentials, without prompting login.
+This is where things can get tricky!
+
+### Ideal case: oauth
+
+_Ideally_, the best way to set this up is with the external service as an OAuth provider,
+though in some cases it works best to use proxy-based authentication like Shibboleth / [REMOTE_USER](https://github.com/cwaldbieser/jhub_remote_user_authenticator).
+The main things to know are:
+
+- Links to `/hub/user-redirect/some/path` will ultimately land users at `/users/theirserver/some/path` after completing login, ensuring the server is running, etc.
+- Setting `Authenticator.auto_login = True` allows beginning the login process without JupyterHub's "Login with..." prompt
+
+_If_ your OAuth provider allows logging in to external services via your oauth provider without prompting, this is enough.
+Not all do, though.
+
+If you've already ensured the server is running, this will _appear_ to the user as if they are being sent directly to their running server.
+But what _actually_ happens is quite a series of redirects, state checks, and cookie-setting:
+
+1. visiting `/hub/user-redirect/some/path` checks if the user is logged in
+   1. if not, begin the login process (`/hub/login?next=/hub/user-redirect/...`)
+   2. redirects to your oauth provider to authenticate the user
+   3. redirects back to `/hub/oauth_callback` to complete login
+   4. redirects back to `/hub/user-redirect/...`
+2. once authenticated, checks that the user's server is running
+   a. if not running, begins launch of the server and redirects to `/hub/spawn-pending/?next=...`
+3. once the server is running, redirects to the actual user server `/users/username/some/path`
+
+Now we're done, right? Actually, no, because the browser doesn't have credentials for their user server!
+This sequence of redirects happens all the time in JpuyterHub launch, and is usually totally transparent.
+
+4. at the user server, check for a token in cookie
+   1. if not present or not valid, begin oauth with the Hub (redirect to `/hub/api/oauth2/authorize/...`)
+   2. hub redirects back to `/users/user/oauth_callback` to complete oauth
+   3. redirect again to the URL that started this internal oauth
+5. finally, arrive at `/users/username/some/path`, the ultimate destination, with valid JupyterHub credentials
+
+The steps that will show users something other than the page you want them to are:
+
+- Step 1.1 will be a prompt e.g. with "Login with..." unless you set `c.Authenticator.auto_login = True`
+- Step 1.2 _may_ be a prompt from your oauth provider. This isn't controlled by JupyterHub, and may not be avoidable.
+- Step 2.2 will show the spawn pending page only if the server is not already running
```

Comment:
Should this refer to Step 2.a?

Reply:
I wanted it to, but myst doesn't understand `2.a` as a nested ol, so it has to be `2.2`.